src/checker.c

```diff
@@ -909,20 +910,15 @@ Type *checkExprTypeEnum(Expr *expr, CheckerContext *ctx, Package *pkg) {
         if (item.init) {
             CheckerContext itemCtx = {.scope = ctx->scope, .desiredType = backingType};
             Type *type = checkExpr(item.init, &itemCtx, pkg);
+            if (itemCtx.mode == ExprMode_Unresolved) goto unresolved; // TODO: @Leak this will leak the 'fields' array
 
             if (!IsConstant(&itemCtx)) {
```

Comment:
We have a b32 expectType(Package *pkg, Type *type, CheckerContext *ctx, Position pos) do you think we should have a b32 expectConstant(Package *pkg, CheckerContext *ctx, Position pos)? This would mean we would get a more consistent error message. What do you think?